lmms_eval/tasks/m3exam/utils.py

```diff
@@ -0,0 +1,895 @@
+from collections import defaultdict
+import re
+import ast
+import random
+import numpy as np
+import os
+import json
+import logging
+# from PIL import Image
+import io
+from lmms_eval.tasks._task_utils.file_utils import generate_submission_file
+from datasets import Image, load_dataset
+
+lmms_logger = logging.getLogger("lmms-eval")
+
+from collections import defaultdict
+import os
+import json
+import argparse
+import csv
+
+
+def generate_one_example(question, lang, method, fill_answer=False):
+    answer_word = {'english': "Answer:", 'chinese': '答案：', 'vietnamese': 'Câu trả lời:', 'thai': 'คำตอบ:', 'italian': 'La risposta:',
+                   'javanese': 'Wangsulan:', 'swahili': 'Jibu:', 'afrikaans': 'Antwoord:' ,'portuguese': 'Responder:'}
+    background = '\n'+'\n'.join(question['background_description']) if question['background_description'] != [] else ''
+    if method == 'default':
+        prompt = background + '\n' + question['question_text'] + '\n' + '\n'.join(question['options']) + f'\n{answer_word[lang]}'
+    elif method == 'en-instruct':
+        prompt = background + '\n' + question['question_text'] + '\n' + '\n'.join(question['options']) + f'\nAnswer:'
+    elif method == 'en-trans':
+        prompt = question['background_description_english'] + '\n' + question['question_text_english'] + '\n' + question['options_english'] + f'\nAnswer:'
+    
+    if fill_answer:
+        prompt += str(question['answer_text'])
+    
+    return prompt
+
+def construct_prompt(doc):
+# (lang, method, setting, model, test_question, dev_question):
+    lang = doc["language"]
+    method = "default"
+    setting = "zero-shot"
```

Comment:
Seems like the setting has been fixed into zero-shot. But I see that you write a lot of if-else statement evaluating the setting is `zero-shot` or not, is it true that there will be more settings in the future?